docs: wallet manager (#1045)
FuelLabs/fuels-ts: apps/docs/src/guide/wallet-manager/getting-started-with-wallet-manager.md

# Getting Started with `WalletManager`

This guide provides step-by-step instructions on how to use `WalletManager`.

## Instantiating `WalletManager`

> The `WalletManager`constructor accepts an optional object to defines its storage. The storage defines how and where the `WalletManager` will store its vaults of wallet. If a storage is not provided, the `WalletManager` uses a default one that does not persist data.

Reviewer: ```suggestion
The `WalletManager` constructor accepts an optional object to define its storage. The storage describes how and where the `WalletManager` will store its vaults of wallets. If storage is not provided, the `WalletManager` uses a default one that does not persist data.
```

For now let's keep it simply and do not worry with the storage. Later we will discuss it more in details.

> To instantiate a `WalletManager` you can simply do as the following code:

Reviewer: ```suggestion
To instantiate a `WalletManager` you can simply:
```

<<< @/../../docs-snippets/src/guide/wallet-manager/getting-started-with-wallet-manager.test.ts#getting-started-with-wallet-manager-1{ts:line-numbers}

## Setting `WalletManager` Password

> By default, a `WalletManager` instance is locked when it's created. Before you can use it, you need to unlock it by setting a password. You can do this by calling the `unlock` method.

Reviewer: ```suggestion
By default, a `WalletManager` instance is locked when created. Before using it, you need to unlock it by setting a password. You can do this by calling the `unlock` method.
```

<<< @/../../docs-snippets/src/guide/wallet-manager/getting-started-with-wallet-manager.test.ts#getting-started-with-wallet-manager-2{ts:line-numbers}

> Once your `WalletManager` is unlocked, it's ready to manage your wallets.

Reviewer: ```suggestion
Once your `WalletManager` is unlocked, it can manage your wallets.
```

## Managing Vaults with WalletManager

A vault in `WalletManager` serves as a secure container for wallets. The `WalletManager` manages wallets by interacting with these vaults, supporting operations such as `getAccounts`, which returns public information about all wallets stored in the vault, and `exportAccount`, which exports a private key for a given wallet address.

> To add a vault, we utilize the `addVault` method. Here's how we can create a Private Key vault and add a private key from a wallet we own:

Reviewer 2: ```suggestion
To add a vault, we utilize the `addVault` method. Here's how we can create a private key vault and add a private key from a wallet we own:
```
Nit

<<< @/../../docs-snippets/src/guide/wallet-manager/getting-started-with-wallet-manager.test.ts#getting-started-with-wallet-manager-3{ts:line-numbers}

> The `addVault` method requires an object with three properties: `type`, `secret`, and `title`. The `WalletManager` currently supports two types of vaults: `privateKeyVault` and `mnemonicVault`. For the `secret`, we use our wallet's private key, and for the `title`, we can provide a custom name.

Reviewer: ```suggestion
The `addVault` method requires an object with three properties: `type`, `secret`, and `title`. The `WalletManager` currently supports two types of vaults: `privateKeyVault` and `mnemonicVault`. We use our wallet's private key as the 'secret' while providing a custom name for the 'title'.
```

By running this code, `WalletManager` creates a new vault instance of the type `privateKey` and adds one account (our wallet) to this newly created vault.

A key feature of the `WalletManager` is its ability to manage multiple vaults, even of the same type. This implies that if you run the `addVault` method again, with the same parameters, `WalletManager` will create another vault of the type `privateKey`, holding the same wallet. Here's an example:

<<< @/../../docs-snippets/src/guide/wallet-manager/getting-started-with-wallet-manager.test.ts#getting-started-with-wallet-manager-4{ts:line-numbers}

After executing this, you will find that your `WalletManager` is managing two `privateKey` vaults, both storing the same wallet.

> Remember, both `title` and `secret` are optional when adding vaults, but providing a `title` makes it easier to manage your vaults and wallets. If you add a vault without providing a `secret`, this will result in one new account (wallet) being generated by the vault it self.

Reviewer: ```suggestion
Remember, both `title` and `secret` are optional when adding vaults, but providing a `title` makes it easier to manage your vaults and wallets. Adding a vault without a `secret` will result in a new account (wallet) being generated by the vault itself.
```

## Using The `WalletManager`

With your `WalletManager` set up, you can now access your vaults and wallets. Here's how to retrieve the details of your vaults:

<<< @/../../docs-snippets/src/guide/wallet-manager/getting-started-with-wallet-manager.test.ts#getting-started-with-wallet-manager-5{ts:line-numbers}

This will output something like this:

<<< @/../../docs-snippets/src/guide/wallet-manager/getting-started-with-wallet-manager.test.ts#getting-started-with-wallet-manager-6{bash:line-numbers}

As you can see, the WalletManager assigns unique `vaultIds` for each vault. The first vault you added has a `vaultId` of `0`, and the second one has a `vaultId` of `1`.

Let's retrieve your wallet instance with the `getWallet` method:

<<< @/../../docs-snippets/src/guide/wallet-manager/getting-started-with-wallet-manager.test.ts#getting-started-with-wallet-manager-7{ts:line-numbers}

> This guide walked through the steps to instantiate a WalletManager, set up its first vault, and retrieve vault information. The following sections will explore more functionalities of WalletManager, examine deeper into the usage of its vaults, and the details of its storage system.

Reviewer: ```suggestion
This guide walked through the steps to instantiate a WalletManager, set up its first vault, and retrieve vault information. The following sections will explore more functionalities of WalletManager, and go deeper into the usage of its vaults and the details of its storage system.
```

Reviewer 2: @arboleya probably also needs ticks for `WalletManager`
